lib/spack/spack/cmd/create.py

```diff
@@ -305,6 +326,7 @@ def __init__(self, name, *args):
     'bazel':      BazelPackageTemplate,
     'python':     PythonPackageTemplate,
     'r':          RPackageTemplate,
+    'perl':       PerlPackageTemplate,
```

Comment:
Can you add `Makefile.PL` and `Build.PL` to `clues` in `BuildSystemGuesser` below? That will allow `spack create` to automatically choose the `PerlPackage` base class. Otherwise, these changes won't actually go into affect.